### PR TITLE
Start daemons on GD restart

### PR DESCRIPTION
**Pull request by kshlm**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -97,6 +98,9 @@ func main() {
 	super.Add(servers.New())
 	addMgmtService(super)
 
+	// Restart previously running daemons
+	daemon.StartAllDaemons()
```
Comment (prashanthpai) [inline main.go]: Some of the rest handlers might have to communicate with these daemons and as soon as the rest server is up, glusterd2 will accept incoming requests. So it'll be safer to start the daemons first and then start supervisor chain of servers.

Comment (kshlm) [reply main.go]: The daemons need the sunrpc server to be up though. I think its acceptable that calls can fail for a little while.

**Comment by kshlm** [comment]
> LGTM 👍 I have one requested one small change.
> It would also be nice to have a e2e test case for this with bricks as the example daemon.

It will be much easier to test this if volume status were working. Without it, I'll need to shell out a lot to test this.

**Comment by prashanthpai** [comment]
retest this please

**Comment by kshlm** [comment]
retest this please

**Comment by kshlm** [comment]
retest this please

**Pull request merged**